OctopusDSC/DSCResources/cTentacleAgent/cTentacleAgent.psm1

```diff
@@ -501,8 +565,22 @@ function Test-TargetResource {
     # Check Ensure value
     if ($Ensure -eq "Present" -and ![string]::IsNullOrEmpty($OctopusServerUrl))
     {
+        if (![string]::IsNullOrEmpty($Space))
+        {
+            # Get reference to the space
+            $spaceRef = Get-Space -Space $Space -ServerUrl $OctopusServerUrl -APIKey $ApiKey
+
+            # Check to see if something was returned
+            if ($null -eq $spaceRef)
+            {
+                # Throw error, space wasn't found
+                throw "Unable to find a space by the name of $Space"
+            }
+        }
+
+
         # Get reference to machine
-        $machine = Get-MachineFromOctopusServer -ServerUrl $OctopusServerUrl -APIKey $ApiKey -Instance $Name
+        $machine = Get-MachineFromOctopusServer -ServerUrl $OctopusServerUrl -APIKey $ApiKey -Instance $Name -SpaceId $spaceRef.Id
```

Comment:
`$spaceRef` could be null here, right? wont this be a null ref exception if the server doesn't support spaces?

Reply:
Tests dont seem to blow up, so think its fine